bootstrap/src/string.ml

```diff
@@ -85,18 +87,23 @@ module Cursor = struct
       bindex: usize;
     }
 
+    let hash_fold t state =
+      state
+      |> hash_fold t.string
+      |> Usize.hash_fold t.bindex
```

Comment:
This syntax is a good example of why I changed the `hash_fold` parameter order.  Absent the `|>` operator this code is much messier and harder to follow.

Reply:
Agreed. I like this way.